bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/handler/IpCameraHandler.java

```diff
@@ -1575,6 +1575,13 @@ void pollCameraRunnable() {
                 setupFfmpegFormat(FFmpegFormat.RTSP_ALARMS);
             }
         }
+        if (ffmpegMjpeg != null) {// check if the thread has frozen due to camera doing a soft reboot
+            localFfmpeg = ffmpegMjpeg;
+            if (localFfmpeg == null || !localFfmpeg.getIsAlive()) {
```

Comment:
`localFfmpeg` cannot be null at this place.

Maybe rather:
```
        localFfmpeg = ffmpegMjpeg;
        if (localFfmpeg != null) {// check if the thread has frozen due to camera doing a soft reboot
            if (!localFfmpeg.getIsAlive()) {
                setupFfmpegFormat(FFmpegFormat.MJPEG);
                logger.debug("MJPEG was not being produced by FFmpeg when it should have been, restarting FFmpeg.");
            }
        }
```

Reply:
fixed and Resolved all comments.